docs/contributors/javascript-testing.md

```diff
@@ -18,3 +27,25 @@ Additionally,
 
 -   `test:update` updates the snapshot tests for components, used if you change a component that has tests attached.
 -   `test:watch` keeps watch of files and automatically re-runs tests when things change.
+
+## How to run end-to-end tests
+
+End-to-end tests are implemented in `tests/e2e-tests/specs/`.
+
+Since these drive the user interface, they need to run against a test environment - i.e. a web server running WordPress, Woo and blocks plugin, with a known state/configuration). 
+
+To set up to run e2e tests:
+
+- `npm run build:e2e-test` to build container images used for e2e
+- `npm run wp-env start` to start the test environment
+
+Then, to run the tests:
+
+- `npm run test:e2e` 
+
+When you're iterating on a new test you'll run this repeatedly until your test is just right.
```

Comment:
Do we know why this is required? I wonder if it might be valuable to add an explanation.

Reply:
I'm assuming this is just a nod to the fact that while tests are in development they might need to be re-run to pick up changes as the test is being developed.

Also, a couple notes (that don't affect this pull, but to keep in mind in future iterations of our test suite):

- ideally we should switch to use `wp-scripts test:e2e` for our tests.
- doing the above will get us access to a number of e2e helpers including: watching changed files and automatically running e2e tests on them, a special debug mode for debugging tests.

Reply:
Clarified this a bit in 6be54268f8404a9c1b610f73d198532fe6824f3b - yes this is just referring to code => test iterate loop :)